packages/libp2p/src/dcutr/dcutr.ts

```diff
@@ -214,9 +253,67 @@ export class DefaultDCUtRService implements Startable {
     }
   }
 
+  /**
+   * This function encapsulates the Simultaneous Connect step (#5) of the [DCUtR protocol](https://github.com/libp2p/specs/blob/master/relay/DCUtR.md#the-protocol)
+   * @param relayedConnection
+   * @param multiaddrs
+   * @param signal
+   */
+  async handleSimultaneousConnect (relayedConnection: Connection, multiaddrs: Multiaddr[], rtt: number, signal: AbortSignal): Promise<void> {
+    logB('waiting for half RTT')
+    // ..and starts a timer for half the RTT measured from the time between
+    // sending the initial Connect and receiving the response
+    // TODO: Find a way to simulate network latency for tests, because rtt is always <= 4ms
```

Comment:
We could have a custom transport for testing that just added latency to every buffer that goes over the connection, but I think the hard thing is simulating different kinds of NAT, and simulating NAT over a network.

The experience of the rust/go folk is that TCP doesn't work as well as QUIC for hole punching so given that we don't have QUIC yet I don't think we should burn a ton of energy on this.